### PR TITLE
Fix: Display EDLP tag applicable to a session

### PR DESCRIPTION
**Pull request by twilkhoo**
## Notion ticket link

[[Registrant Experience] Dates to select session are incorrect](https://www.notion.so/uwblueprintexecs/Registrant-Experience-Dates-to-select-session-are-incorrect-ee341f9de21d421eb8be0bc903ea4843)

## Implementation description
* Passed in a particular session as a prop
* Using the session's dates, I can conditionally render the EDLP tags if the camper's selected EDLP dates are within the particular session's range

## Steps to test
1. Go to http://localhost:3000/admin/camp/6396319f01dd5e987a03b516
2. Ensure that the campers that have EDLP only display the purple EDLP tags if the EDLP date is in range.

**Before:**
image

**After:**
image

## What should reviewers focus on?
* Correctness, functionality.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx [hunk 1]
```diff
@@ -131,7 +130,6 @@ const EditCamperModal = ({
         <ModalHeader>
           Edit {camper.firstName} {camper.lastName}&apos;s Information
         </ModalHeader>
-        <ModalCloseButton />
```
Comment (rchang0501) [inline frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx:134]: just wanted to confirm the reason for removing this button does it just not match the designs?

File: frontend/src/components/pages/CampOverview/ViewCamperModal/PickupDropoffTableRow.tsx [hunk 2]
```diff
@@ -54,7 +54,7 @@ const PickupDropoffTime = ({
   return (
     <Td pl={0} pt={2} pb={0} color={color}>
       <HStack>
-        {pickup ? <SunriseIcon fill={color} /> : <SunsetIcon fill={color} />}
```
Comment (rchang0501) [inline frontend/src/components/pages/CampOverview/ViewCamperModal/PickupDropoffTableRow.tsx:57]: we need to revert this, the pickup icon is supposed to be the sunrise icon not the sunset one (I know it's kinda confusing)

Comment (rchang0501) [reply frontend/src/components/pages/CampOverview/ViewCamperModal/PickupDropoffTableRow.tsx:57]: Expected look:

Screen Shot 2023-01-28 at 2 12 05 PM

File: frontend/src/utils/CampUtils.ts [hunk 3]
```diff
@@ -483,3 +483,26 @@ export const getFormattedCampDateRange = (camp: CampResponse): string => {
     ? `${format(firstCampDate, "PP")} - ${format(lastCampDate, "PP")}`
     : "";
 };
+
+export const checkDatesInRange = (
```
Comment (rchang0501) [inline frontend/src/utils/CampUtils.ts:487]: I think we can simplify this function a bit - smthn like this: 

```
export const checkDatesInRange = (
  rangeDates: string[],
  datesToCheck?: Date[],
): boolean => {
  if (!datesToCheck || datesToCheck.length === 0) return false;

  const minDate = Math.min(...rangeDates.map((date) => Date.parse(date)));
  const maxDate = Math.max(...rangeDates.map((date) => Date.parse(date)));

  return datesToCheck.some(
    (date: Date) =>
      new Date(date).getTime() >= minDate &&
      new Date(date).getTime() <= maxDate,
  );
};
```

File: frontend/src/components/pages/CampOverview/ViewCamperModal/PickupDropoffTableRow.tsx [hunk 4]
```diff
@@ -54,7 +54,7 @@ const PickupDropoffTime = ({
...
+        {pickup ? <SunsetIcon fill={color} /> : <SunriseIcon fill={color} />}
```
Comment (rchang0501) [inline frontend/src/components/pages/CampOverview/ViewCamperModal/PickupDropoffTableRow.tsx:57]: I believe need to revert this - pickup should be the sunrise button (I know it's kinda confusing)

Expected look: 
Screen Shot 2023-01-28 at 2 12 05 PM

Design ref:
Screen Shot 2023-01-28 at 2 13 30 PM

**Review by rchang0501 — CHANGES_REQUESTED** [review]
overall looks good! 

left 2 comments and one clarification question - lmk if you have any questions!

File: frontend/src/components/pages/CampOverview/ViewCamperModal/PickupDropoffTableRow.tsx (hunk 4 above)
Comment (twilkhoo) [reply frontend/src/components/pages/CampOverview/ViewCamperModal/PickupDropoffTableRow.tsx:57]: Sounds good, I'll revert it but I think I'll make a lower priority ticket just going over the inconsistencies of these icons. 
image
Here, early dropoff is sunrise and late pickup is sunset (which IMO, makes a lot more sense, but we can verify with the designers)

File: frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx (hunk 1 above)
Comment (twilkhoo) [reply frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx:134]: Yeah, the modals don't have the close button (consistency).

File: frontend/src/utils/CampUtils.ts (hunk 3 above)
Comment (twilkhoo) [reply frontend/src/utils/CampUtils.ts:487]: Works like a charm, thanks!

File: frontend/src/components/pages/CampOverview/ViewCamperModal/PickupDropoffTableRow.tsx (hunk 4 above)
Comment (rchang0501) [reply frontend/src/components/pages/CampOverview/ViewCamperModal/PickupDropoffTableRow.tsx]: yep sounds good, i totally thought it was the other way around initially as well but I feel like if we are gonna make a change we should choose the easier one to change (ie the most consistent set we have rn)

**Review by rchang0501 — APPROVED** [review]
lgtm!

**Pull request merged**